pkg/thumbnails/imgsource/imgsource.go

```diff
@@ -0,0 +1,33 @@
+package imgsource
+
+import "image"
+
+// Source defines the interface for image sources
+type Source interface {
+	Get(path string, ctx SourceContext) (image.Image, error)
+}
+
+// NewContext creates a new SourceContext instance
+func NewContext() SourceContext {
```

Comment:
have you considered using `context.Context` here? You can still store key-values on a typed and thread safe manner 🗡